packages/devextreme/js/__internal/core/state_manager/state_manager.ts

```diff
@@ -0,0 +1,199 @@
+import { Logger } from './logger';
+import { PreactSignalValueContainerManagerFactory } from './preact_signal_value_container_manager';
+// eslint-disable-next-line spellcheck/spell-checker
+import { ReduxDevToolsConnector } from './redux_dev_tools_connector';
+import type * as StateManagementTypes from './types';
+import { deepCopy, joinStatePath } from './utils';
+
+class StateManager implements StateManagementTypes.StateManager {
+  private readonly devToolsConnector: StateManagementTypes.DevToolsConnector;
+
+  private readonly logger: StateManagementTypes.Logger;
+
+  private componentState: StateManagementTypes.ComponentState;
```

Comment:
Let's add `readonly` keyword as u did for other fields
```suggestion
  private readonly componentState: StateManagementTypes.ComponentState;
```

Reply:
Changed